go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -412,6 +412,27 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 		return nil
 	}
 
+	activeNetwork, err := bsnc.getActiveNetworkForNamespace(pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed looking for the active network at namespace '%s': %w", pod.Namespace, err)
+	}
+
+	on, networkMap, err := util.GetPodNADToNetworkMappingWithActiveNetwork(pod, bsnc.NetInfo, activeNetwork)
+	if err != nil {
+		bsnc.recordPodErrorEvent(pod, err)
+		// configuration error, no need to retry, do not return error
+		klog.Errorf("Error getting network-attachment for pod %s/%s network %s: %v",
+			pod.Namespace, pod.Name, bsnc.GetNetworkName(), err)
+		return nil
+	}
+
+	if !on {
+		// the pod is not attached to this specific network
+		klog.V(5).Infof("Pod %s/%s is not attached on this network controller %s",
+			pod.Namespace, pod.Name, bsnc.GetNetworkName())
+		return nil
```

Comment:
so for L3/localnet I don't see a problem.. for L2 we usually create a remote port for the remote pod, so if we move it here will we not call `pInfo, err := bsnc.deletePodLogicalPort(pod, portInfoMap[nadName], nadName)` in that case? or are the remote ports for pods in l2 removed elsewhere...

Reply:
this code will remove that port as well I think, but I dont see how that is related to the change here. The remote pod would still be on this network right?

Reply:
my bad. I was totally confused between local/remote pods and
same/notsame networks.

this fix is trying to skip processing pods that don't belong to this network.